src/Compute/Compute/Images/GetAzureVMImageCommand.cs

```diff
@@ -95,7 +102,69 @@ public override void ExecuteCmdlet()
 
             ExecuteClientAction(() =>
             {
-                if (this.ParameterSetName.Equals(ListVMImageParamSetName) || WildcardPattern.ContainsWildcardCharacters(Version))
+                if ((this.IsParameterBound(c => c.EdgeZone)) && this.EdgeZone != null)
+                {
+                    if (this.ParameterSetName.Equals(ListVMImageParamSetName) || WildcardPattern.ContainsWildcardCharacters(Version))
+                    {
+                        var result = this.VirtualMachineImagesEdgeZoneClient.ListWithHttpMessagesAsync(
+                            this.Location.Canonicalize(),
+                            this.EdgeZone.Canonicalize(),
+                            this.PublisherName,
+                            this.Offer,
+                            this.Skus,
+                            top: this.Top,
+                            orderby: this.OrderBy
+                            ).GetAwaiter().GetResult();
+
+                        var images = from r in result.Body
+                                     select new PSVirtualMachineImage
+                                     {
+                                         RequestId = result.RequestId,
+                                         StatusCode = result.Response.StatusCode,
+                                         Id = r.Id,
+                                         Location = r.Location,
+                                         EdgeZone = r.ExtendedLocation.Name,
+                                         Version = r.Name,
+                                         PublisherName = this.PublisherName,
+                                         Offer = this.Offer,
+                                         Skus = this.Skus
+                                     };
+
+                        WriteObject(SubResourceWildcardFilter(Version, images), true);
+                    }
+                    else
+                    {
+                        var response = this.VirtualMachineImagesEdgeZoneClient.GetWithHttpMessagesAsync(
+                            this.Location.Canonicalize(),
+                            this.EdgeZone,
+                            this.PublisherName,
+                            this.Offer,
+                            this.Skus,
+                            version: this.Version).GetAwaiter().GetResult();
+
+                        var image = new PSVirtualMachineImageDetail
+                        {
+                            RequestId = response.RequestId,
```

Comment:
usually mapping of returned property from the HTTP call to PS object is done like this using mapper: https://github.com/Azure/azure-powershell/blob/master/src/Compute/Compute/Disk/NewAzDiskAccessCommand.cs#L57

I know this cmdlet was already like this, but I suggest using mapper instead going forward